caproto/_state.py

```diff
@@ -22,70 +22,77 @@
 COMMAND_TRIGGERED_CIRCUIT_TRANSITIONS = {
     CLIENT: {
         SEND_VERSION_REQUEST: {
-            EchoRequest: SEND_VERSION_REQUEST,
-            EchoResponse: SEND_VERSION_REQUEST,
             VersionRequest: AWAIT_VERSION_RESPONSE,
-            VersionResponse: CONNECTED,
```

Comment:
A lot of complexity in the states came from bouncing directly to CONNECTED, which is in correct in my view. If the CLIENT is in a SEND_VERSION_REQUEST state, it should still send a version request (and then get a response in turn).